pkg/build/buildkit/errors.go

```diff
@@ -49,54 +47,51 @@ func (e CommandErr) Error() string {
 	return fmt.Sprintf("error on stage %s: %s", e.Stage, e.Err.Error())
 }
 
-// GetErrorMessage returns the parsed error message
-func GetErrorMessage(err error, tag string) error {
+// GetSolveErrorMessage returns the parsed error message
+func GetSolveErrorMessage(err error) error {
 	if err == nil {
 		return nil
 	}
 
-	imageCtrl := registry.NewImageCtrl(okteto.Config{})
-	imageRegistry, imageTag := imageCtrl.GetRegistryAndRepo(tag)
+	imageFromError := extractImageFromError(err)
 	switch {
-	case isLoggedIntoRegistryButDontHavePermissions(err):
-		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to push image '%s': You are not authorized to push image '%s'", tag, imageTag),
-			Hint: fmt.Sprintf("Please log in into the registry '%s' with a user with push permissions to '%s' or use another image.", imageRegistry, imageTag),
-		}
-	case isNotLoggedIntoRegistry(err):
-		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to push image '%s': You are not authorized to push image '%s'", tag, imageTag),
-			Hint: fmt.Sprintf("Log in into the registry '%s' and verify that you have permissions to push the image '%s'.", imageRegistry, imageTag),
-		}
 	case isBuildkitServiceUnavailable(err):
 		err = oktetoErrors.UserError{
 			E:    fmt.Errorf("buildkit service is not available at the moment"),
 			Hint: "Please try again later.",
 		}
-	case isPullAccessDenied(err):
-		if imageTag == "" {
-			imageTag = extractImageTagFromPullAccessDeniedError(err)
-		}
+	case isLoggedIntoRegistryButDontHavePermissions(err) ||
+		isNotLoggedIntoRegistry(err) ||
+		isPullAccessDenied(err) ||
+		isNotFound(err) ||
+		isHostNotFound(err):
 		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to pull image '%s'. The repository is not accessible or it does not exist", imageTag),
-			Hint: fmt.Sprintf("Please verify the name of the image '%s' to make sure it exists.", imageTag),
+			E: fmt.Errorf("the image '%s' is not accessible or it does not exist", imageFromError),
```

Comment:
We were distinguishing if the error was a permission problem vs not logged into the registry. Do we want to change that?

Reply:
That is the case, yes